pyaedt/modules/PostProcessor.py

```diff
@@ -4911,3 +4915,455 @@ def sample_ami_waveform(
+                    quantity, ", ".join(AVAILABLE_QUANTITIES)
+                )
+            )
+        self.calculations.append(
+            [entity, geometry, geometry_name, quantity, normal, side, mesh, ref_temperature, False]
+        )  # TODO : last argument not documented
+
+    @pyaedt_function_handler()
+    def get_field_summary_data(self, sweep_name=None, design_variation={}, intrinsic_value="", pandas_output=False):
+        with tempfile.NamedTemporaryFile(mode="w+", delete=False) as temp_file:
+            temp_file.close()
+            self.export_csv(temp_file.name, sweep_name, design_variation, intrinsic_value)
+            with open(temp_file.name, "r") as f:
+                for _ in range(4):
+                    _ = next(f)
+                reader = csv.DictReader(f)
+                out_dict = defaultdict(list)
+                for row in reader:
+                    for key in row.keys():
+                        out_dict[key].append(row[key])
+            os.remove(temp_file.name)
+            if pandas_output:
+                if pd is None:
+                    raise ImportError("pandas package is needed.")
+                return pd.DataFrame.from_dict(out_dict)
+        return out_dict
+
+    @pyaedt_function_handler()
+    def export_csv(self, filename, sweep_name=None, design_variation={}, intrinsic_value=""):
+        if not sweep_name:
+            sweep_name = self._app.nominal_sweep
+        dv_string = ""
+        for el in design_variation:
+            dv_string += el + "='" + design_variation[el] + "' "
+        self._create_field_summary(sweep_name, dv_string)
+        self._app.osolution.ExportFieldsSummary(
+            [
+                "SolutionName:=",
+                sweep_name,
+                "DesignVariationKey:=",
+                dv_string,
+                "ExportFileName:=",
+                filename,
+                "IntrinsicValue:=",
+                intrinsic_value,
+            ]
+        )
+        return True
+
+    @pyaedt_function_handler()
+    def _create_field_summary(self, setup, variation):
+        arg = ["SolutionName:=", setup, "Variation:=", variation]
+        for i in self.calculations:
+            arg.append("Calculation:=")
+            arg.append(i)
+        self._app.osolution.EditFieldsSummarySetting(arg)
+
+
+class IcepakPostProcessor(PostProcessor, object):
+    def __init__(self, app):
+        PostProcessorCommon.__init__(self, app)
+
+    @pyaedt_function_handler()
+    def create_field_summary(self):
+        return FieldSummary(self._app)
+
+    @pyaedt_function_handler()
+    def get_fans_operating_point(self, export_file=None, setup_name=None, timestep=None, design_variation=None):
+        """
+        Get the operating point of the fans in the design.
+
+        Parameters
+        ----------
+        export_file : str, optional
+            Name of the file to save the operating point of the fans. The default is
```

Comment:
```suggestion
            Name of the file to save the operating point of the fans to. The default is
```
Sorry I missed adding the "to" earlier. Please search and add where necessary.